rest/model/codegen.go

```diff
@@ -0,0 +1,178 @@
+package model
+
+import (
+	"bytes"
+	"io/ioutil"
+	"regexp"
+	"sort"
+	"strings"
+	"text/template"
+
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	gqlparser "github.com/vektah/gqlparser/v2"
+	"github.com/vektah/gqlparser/v2/ast"
+	"golang.org/x/tools/imports"
+)
+
+const (
+	fileTemplate   = "templates/file.gotmpl"
+	structTemplate = "templates/struct.gotmpl"
+	fieldTemplate  = "templates/field.gotmpl"
+)
+
+type fileInfo struct {
+	Package string
+	Structs string
+	Code    string
+}
+
+type structInfo struct {
+	Name   string
+	Fields string
+}
+
+type fieldInfo struct {
+	Name    string
+	Type    string
+	JsonTag string
+}
+
+func SchemaToGo(schema string) ([]byte, error) {
+	source := ast.Source{
+		Input: schema,
+	}
+	fileTemplate, err := getFileTmpl()
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading file template")
+	}
+	structTemplate, err := getStructTmpl()
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading struct template")
+	}
+	fieldTemplate, err := getFieldTmpl()
+	if err != nil {
+		return nil, errors.Wrap(err, "error loading field template")
+	}
+	parsedAst, parseErr := gqlparser.LoadSchema(&source)
+	if parseErr != nil {
+		return nil, errors.New(parseErr.Error())
```

Comment:
I would expect `Error()` to do the same thing as string formatting, so you can just `errors.Wrap(err, "helpful text")` (and rename `parseErr` to `err`).

Reply:
Oh it was a different type but I didn't realize it implemented the error interface. Fixed